src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/InstanceDiscovery.cs

```diff
@@ -187,11 +187,15 @@ private async Task DiscoverAsync(Uri authority, bool validateAuthority, RequestC
                 // All other unexpected exceptions will still bubble up, as always.
                 if (validateAuthority)
                 {
-                    // hard stop here
-                    throw new AdalException(
-                        (ex.ErrorCode == "invalid_instance")
-                            ? AdalError.AuthorityNotInValidList
-                            : AdalError.AuthorityValidationFailed, ex);
+                    if (Equals(ex.ErrorCode == ErrorCodes.InvalidInstance, StringComparison.OrdinalIgnoreCase))
+                    {
+                        throw new AdalException(
+                            AdalError.AuthorityNotInValidList, 
+                            AdalError.AuthorityValidationFailed, 
+                            ex);
+                    }
+                    throw;
+                    //throw new AdalException(ex.Response.StatusCode.ToString(), ex.InnerException.Message);                   
```

Comment:
remove dead code.

Reply:
oops. yes. sorry about that.

---
In reply to: [356264930](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1682#discussion_r356264930) [](ancestors = 356264930)